nidirect_landing_pages/nidirect_landing_pages.module

```diff
@@ -52,9 +175,50 @@ function nidirect_landing_pages_form_alter(&$form, FormStateInterface $form_stat
       $msg .= "As an example, if you select 'Motoring / Road Safety' here then when you visit the page at ";
       $msg .= "/information-and-services/motoring there will be a link to this page labelled 'Road Safety'.";
       $form['field_subtheme']['widget']['#description'] = t($msg);
+    }
+    // Only show 'Listing' field if 'manually control listing'
+    // checkbox has been checked.
+    $form['field_listing']['#states'] = ['visible' => [':input[name="field_manually_control_listing[value]"]' => ['checked' => TRUE]]];
+    // Add ajax to the 'manually control listing' checkbox.
+    if (isset($form['field_manually_control_listing'])) {
+      $form['field_manually_control_listing']['widget']['value']['#ajax'] = [
+        'callback' => 'nidirect_landing_pages_populate_listing',
+        'event' => 'change',
+        'wrapper' => 'edit-field-listing-wrapper',
+      ];
+    }
+    // Add a validate handler to detect any duplicates.
+    $form['#validate'][] = 'nidirect_landing_pages_form_validate';
+  }
+}
 
-      // Add a validate handler to detect any duplicates.
-      $form['#validate'][] = 'nidirect_landing_pages_landing_page_form_validate';
+/**
+ * Utility Ajax function.
+ */
+function nidirect_landing_pages_populate_listing(array &$form, FormStateInterface $form_state) {
+  // The main purpose of this is to make sure that
+  // nidirect_common_node_prepare_form is called.
+  if (!$form_state->getValue('field_manually_control_listing')['value']) {
+    // If 'manual control' checkbox is unchecked, make sure
+    // that listing field is not shown.
+    $form['field_listing']['#access'] = FALSE;
```

Comment:
Do you want to visually hide it or remove it from the form payload altogether? I've found this approach can cause a headache if you're using the value later on as `#access` can be too severe. If you do need it later, changing to hidden might be better.

Reply:
very good point, have now used hidden and it does the job